gcc/config/riscv/riscv.cc

```diff
@@ -11470,10 +11471,14 @@ riscv_override_options_internal (struct gcc_options *opts)
       = (cf_protection_level) (opts->x_flag_cf_protection | CF_SET);
     }
 
+  if (opts->x_riscv_mno_compress)
+    opts->x_riscv_zc_subext = 0;
+
   /* Enable advanced fusion for the ARC-V cores that support it unless
      explicitly disabled.
      TODO Add arcv_rmx500 once it supports fusion.  */
   if ((riscv_microarchitecture == arcv_rhx100
+        || riscv_microarchitecture == arcv_rmx500			  
```

Comment:
```suggestion
    explicitly disabled. */
if ((riscv_microarchitecture == arcv_rhx100 
```
It was decided that the mtune for rmx500 shouldn't imply the advanced fusions. They have to be explicitly turned on with the option.

Reply:
I have let the advance fusions to take place only under the explicit flag